### ISSUE TITLE
[resize-observer] svg interaction

### ISSUE THREAD

**Issue by bkardell** [w3c/csswg-drafts#4032]
ResizeObserver mentions SVG specially several times (see, for example https://drafts.csswg.org/resize-observer/#calculate-box-size).  I don't recall the intent of this, but I am wondering if perhaps it is intended for observing the resizing of things _inside_ the SVG and not the SVG itself?  It seems that if I had some construct with a svg with `width="100%"` the SVG itself is resizing, but if it is observing the boundingBox then those things will not be reported.  This [little pen](https://codepen.io/bkardell/pen/RzwNmv) illustrates what I mean - here, I believe the path should fill green, but it doesn't because it seems that what is being observed is the bounding box.  The pen also updates the reporting of the size of that bounding box every 100ms and as you can see, it never updates.

This seems strange and it does seem (at least from what I can tell) implemented this way in all of the browsers.  As an author, it's true that you can work around this by wrapping a div around it and monitoring that instead, but that seems like a strange workaround and like maybe this creates a need for special understanding.  It seems what we want to observe here is not this?

**Comment by AmeliaBR** [comment]
For SVG elements that have associated CSS layout boxes, you should definitely be able to access the CSS content and border boxes, the same as on an `<img>` or container element. This applies to the root `<svg>` in an SVG document and any inline `<svg>` element that is a direct child of HTML elements (or other non-SVG elements) or a direct child of an SVG `<foreignObject>` element, which creates an internal CSS layout context.  Other specifications have special rules for SVG elements with or without CSS boxes, including [`transform-origin`(https://drafts.csswg.org/css-transforms/#transform-origin-property) and some [CSSOM View APIs](https://drafts.csswg.org/cssom-view/#svg-layout-box).

For SVG elements _without_ CSS layout boxes (i.e, shape or group elements, or `<svg>` nested inside other `<svg>`), then `content-box` and `border-box` aren't defined. So perhaps it makes sense to default to the SVG [object bounding box](https://svgwg.org/svg2-draft/coords.html#BoundingBoxes) size.  This will only be useful in limited cases — where the container `<svg>` does not have a `viewBox` attribute and this element or its children are defined using percentage dimensions, which will therefore change as the container is resized — but most other cases could be handled by listening for resize events on the SVG that _does_ have CSS layout boxes, _if_ those are handled as normal CSS boxes.

**Comment by gregwhitworth** [comment]
> For SVG elements that have associated CSS layout boxes, you should definitely be able to access the CSS content and border boxes, the same as on an <img> or container element. This applies to the root <svg> in an SVG document and any inline <svg> element that is a direct child of HTML elements (or other non-SVG elements) or a direct child of an SVG <foreignObject> element, which creates an internal CSS layout context.

I completely agree that this is what an author would expect and may just be an oversight, I'm personally in favor to adjusting the spec text to allow for this scenario to work. FYI @atotic

**Comment by AmeliaBR** [comment]
SVG  bug for discussing what should happen for elements _without_ CSS layout boxes: https://github.com/w3c/svgwg/issues/713
